### ISSUE TITLE
[Request] Add OpenSSL

### ISSUE THREAD

**Issue by CoalaJoe** [EventGhost/EventGhost#185]
I tried to write a WebSocket Client wich communicates via WAMP to a Server. The Server uses SSL and I want to use that.

**Comment by kdschlosser** [comment]
I can swear this has been talked about before in the forum. You might want to search in there on how to do it. i do not recall if there was a way or not. but let me know what you find out.

**Comment by topic2k** [comment]
Maybe [this thread](http://www.eventghost.net/forum/viewtopic.php?f=4&t=7315&hilit=OpenSSL) in the forum helps you.

Do you write a plugin for EG or are just using Python-Script actions from within EG?

**Comment by CoalaJoe** [comment]
I am writing something with autobahnJS and could then publish it as a plugin.

**Comment by topic2k** [comment]
If you make a plugin, needed modules can be placed in the plugin folder, else you could try to copy the modules in the <install-path>\EventGhost\lib27\site-packages folder.

**Comment by CoalaJoe** [comment]
Yeah, I did that. But it seems like there has to be openSSL compiled and installed at the client.

**Comment by topic2k** [comment]
Please, can you give some code to demonstrate the problem?

**Comment by CoalaJoe** [comment]
I can provid it in 3 - 5 days. I am out of office.

**Comment by kdschlosser** [comment]
yeah there are modules that will not install just simply copying them point in fact pyQt4 is one that is like that as well.. if added when EG is compiled it runs fine..  I have encountered this problem before and that is one of the reasons why i tried to push to have the package manager added and to allow it to be added to the eg.RegisterPlugin. I still have the code on it if you wanted to take a look Mr.T LOL

**Comment by CoalaJoe** [comment]
Sorry, the code is already missing. But I think it is not so important. The problem was/is that I use an encryption and the openSSL is not provided in the package or windows installation.

**Comment by kdschlosser** [comment]
Try installing python 2.7 32 bit and then install the open SSL package into that python when EG runs it will automatically add any packages that have been added to that interpreter to EventGhost.. see if that may work for ya.

I am trying to think of the simplest solution for ya. if that doesn't work then i can walk ya through how to build it. there is a tutorial on how to build EG in the forums under the tutorial section. but you would have to make a change to make sure the module gets included into the build.

**Comment by CoalaJoe** [comment]
I will try it tomorrow. Thanks for the support!

**Comment by kdschlosser** [comment]
yeah let me know.... all you would have to do it install python.. go to your python/Scripts folder and open a command line windows

and in that window key in

    pip install openSSL

and then run EventGhost after it is done and see if it works

**Comment by CoalaJoe** [comment]
Can I run EventGhost with another Python installation?

**Comment by kdschlosser** [comment]
OK let me explain how this will work fully...

EG version 0.4 cannot do this only EG 0.5 can. and any version of EG will run along side of python.

You cannot replace the python version that EG is using without great code changes. so squash that one.

EG 0.5 has the ability built into it to load the modules from an external python installation.. Now to make sure you do not have any issues I would install python 2.7.9 32bit.. this is the same python version that EG uses internally. so if any modules you install this will make sure that if they set things up dependent on the python version and bits it will match what EG is using..

and to use the module all you would have to do is to import it like any other module

**Comment by CoalaJoe** [comment]
Nice feature!

Thanks. I'll look into this.

**Comment by CoalaJoe** [comment]
Is there any estimated stable release date of 0.5?

**Comment by kdschlosser** [comment]
ummm... it's pretty stable now.. I haven't had any big issues with it.. there is only one issue at present that is a hang up because of thing else going on I cann also tell you hot to fix that one issue.. the building system for some reason is not functioning properly to allow the change i made to fix it to be built. but if you take the previous build and then delete one line in one file it's good to go.

https://ci.appveyor.com/api/buildjobs/9ocm8nip66yn75mu/artifacts/_build%2Foutput%2FEventGhost_0.5.0-beta6_Setup.exe

that's the link to the most recent build.
and after installing if you go into the folder YOUR_INSTALL_DIR/eg/Classes folder
YOUR_INSTALL_FOLDER being the location you chose to install it

and edit the file SpinNumCtrl.py with a text editor other then notepad. notepad++ works or sublime text. go to line 98 which will be numCtrl.SetMin(minValue) and delete the line it will be good to go.

**Comment by CoalaJoe** [comment]
Well. Seems to work well, I installed that package.
 
The only problem I have is that I work in a team with high expectations. They do not like it when software is labeled as "BETA", we would like to deploy the appliction to hounderts of PCs.

Maybe we can talk more privately than Github-Issues?

**Comment by kdschlosser** [comment]
Give me a couple of hours and I will get back to ya.

**Comment by topic2k** [comment]
@CoalaJoe : You can use the [forum](www.eventghost.net/forum) or i can invite you to [slack](https://eventghost-slackin.herokuapp.com/).

**Comment by kdschlosser** [comment]
yeah the slack thing would be good.

**Comment by CoalaJoe** [comment]
@topic2k @kdschlosser Thanks. I'll join you there.
